### ISSUE TITLE
Provide additional syntax groups (heex tags, heex components, `~p` routes)

### ISSUE THREAD

**Issue by sodapopcan** [phoenixframework/tree-sitter-heex#36]
Hey hey!

Would it be possible to give `<%=` and `%>` their own highlight group (apologies, I'm using vim terminology here, I'm not sure what treesitter calls them as I'm brand new to it)?  As it stands, all characters are `{ 'tag.delimiter' }`.  It would be nice to be able to be able to highlight heex delimiters separately from HTML tags.

Similarly heex component calls (`<.foo />`) are simply `{ 'function' }`, though I would love to be able to style them independently.  Also, would it not make more sense for them to be `function.call` instead of `function`?

Finally, the static part of verified routes `~p"/this/part"` belongs to no group whatsoever.

I'm way out of my depth with C but I'd willing to try my hand at any of these if they are acceptable changes.

Thank you so much!

**Comment by chrismccord** [comment]
Please give it a shot and we'd be happy to review!

**Comment by the-mikedavis** [comment]
I'm not sure how nvim-treesitter works with Vim highlight groups but I believe the changes for this will need to be made against the [`nvim-treesitter` highlight queries](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/queries/heex/highlights.scm) rather than this repo. The parser here emits nodes like `<%=` and `(text)` which are pattern-matched in those queries with captures like `@tag.delimiter` and `@text`.

**Comment by sodapopcan** [comment]
@the-mikedavis Ah, interesting.  Thanks for the info, I may open an issue there.

**Comment by sodapopcan** [comment]
So I reported this a little early before exploring more deeply.  I was using `:lua print (vim.inspect(vim.treesitter.get_captures_at_cursor(0)))` to inspect what the highlight groups were (which I found a quick search after realizing `zS` from scriptease does not work) and it wasn't giving me the whole picture.  It turns out there is `:Inspect` built right into treesitter which gives more information... who knew! (Well, lots of people other than me knew.)

So Heex delimiters are actually `@tag.delimiter.heex` and components are actually `@function.heex`.  The only unfortunate thing here is that _all_ `<` and `>` are `@tag.delimiter.heex`, but I can look into that along with `~p` static paths which indeed do not belong to any group.
